wordpress-mobile/WordPress-Android
# Add search to media picker

Fixes #12637 

This PR adds search capability to the Media picker. The search filters out the results by name from the currently showing items. The search should survive screen rotation. 

To test:
- Go to Media library/+/Choose from device
- Notice there is the search icon in the toolbar
- Click on the search icon
- The search field expands
- Rotate your device
- The search field is still expanded
- Input search query
- Notice that the results get filtered out
- Delete the query
- The results reappear
- Select an item in the search state
- The action mode appears over the search
- Confirm selection
- Selected item is uploaded

PR submission checklist:
